docs/web_advanced.rst

```diff
@@ -403,6 +404,16 @@ This is required for type safety (e.g. when checking with mypy)::
         data = request.app[my_private_key]
         # reveal_type(data) -> str
 
+.. note::
+
+   If warnings are explicitly enabled (e.g. ``-We`` or ``-Wall``) then
+   using :class:`str` keys will trigger a warning. Like other warnings
+   this can be filtered out easily if you have no interest in this feature.
+   e.g. Add ``-W 'ignore:It is recommended to use web.AppKey instances for keys.'``
```

Comment:
I usually prefer to be as narrow as possible when making such exclusions:
```suggestion
   e.g. Add ``-W 'ignore:It is recommended to use web.AppKey instances for keys.:aiohttp.web_exceptions.NotAppKeyWarning:aiohttp.web_app'``
```
(although I'm not sure that the module part of the spec is right, because of the `stacklevel`, maybe drop it from my suggestion)

Can we advertise this? Besides, it's an “advanced” document.

Reply:
I'm afraid it's not valid either way, otherwise I would have just used the type alone which is perfectly accurate: https://github.com/python/cpython/issues/66733

I'm tempted to not merge this in, as nobody else has reported any concerns. The original report was from someone using 3.9.0b0, which wasn't filtering out the warnings by default. Nobody who explicitly enabled warnings has expressed any concerns with the presence of these warnings.

Reply:
It's not? For some, I was convinced that I saw this work in cases when importing the exception wasn't causing it...

Reply:
It basically only works when you manipulate the PYTHONPATH beforehand: https://github.com/python/cpython/issues/66733#issuecomment-1093665963
Otherwise, the warnings filters are resolved before the import paths are resolved. Requiring PYTHONPATH manipulation is not a robust enough experience to recommend in the docs.

Reply:
I'll close this for now, as I've not seen any other user reports. Will reopen it if anybody reports any concerns over the warning.